src/util/data.js

```diff
@@ -186,6 +186,30 @@ export async function fetchVariables(ownertype, ref, all, signal) {
   return await fetch(apiurl(path), null, signal);
 }
 
+export async function createRemoteSource(
+  token, type, name, clientID, clientSecret, apiURL, authType, registrationEnabled,
+  loginEnabled, signal,
+) {
+  let path = "/remotesources";
+  let init = {
+    method: "POST",
+    body: JSON.stringify({
+      name,
+      apiurl: apiURL,
+      type,
+      auth_type: authType,
+      skip_verify: false, // what is that for?
+      oauth_2_client_id: clientID,
+      oauth_2_client_secret: clientSecret,
+      //ssh_host_key: undefined, // not used...
+      skip_ssh_host_key_check: true,
```

Comment:
I won't set this to true by default since it will create security issues (like man in the middle attacks). Users should define the remote source ssh host key.

Reply:
3) What do you think about having only one checkbox (for `skip_verify`) and the field to set the SSH Host Key. That way, `skip_ssh_host_key_check` would be false unless the SSH Key was not provided (taking in consideration GitHub default key). I could also add a message with the recommendation to use a SSH Key.

Reply:
I'll prefer to force the user to explicitly set `skip_ssh_host_key_check` due to the security concerns related to skipping the check.

Reply:
> I'll prefer to force the user to explicitly set `skip_ssh_host_key_check` due to the security concerns related to skipping the check.

Ok. 
1) Is it ok to let the user skip the SSH key check only if the SSH key is not set?
2) Should I add a message asking for confirmation if it is set to skip?

Reply:
> 1. Is it ok to let the user skip the SSH key check only if the SSH key is not set?

Currently in the cli you can pass both a ssh key and also set `skip ssh host key check`, then the ssh key won't be used but saved anyway. If you update the remove source by setting `skip ssh host key check` to false, then the currently saved one will be used. So we could do the same leaving both fields available without graying out anything.

> 2\. Should I add a message asking for confirmation if it is set to skip?

If 1 is ok for you, then no. Leave it simple. You already did a great work. We could improve it later if we found some edge cases.

Reply:
Ok. I'm going just to do some final tests and let you when its good to merge.

Should I also add default values for GitLab? The API: https://api.gitlab.com and the SSH Key (I guess I can get it from my ~/.ssh/known_hosts but there's probably a better source).

Reply:
I've been using it for the last days. It is working good, the only thing I had problems with was the SSH Host Key.

When creating the GitHub source with the Skip SSH Host Key Check disable, the pipeline failed in the clone stage.

Reply:
> I've been using it for the last days. It is working good, the only thing I had problems with was the SSH Host Key.
> 
> When creating the GitHub source with the Skip SSH Host Key Check disable, the pipeline failed in the clone stage.

If it failed then probably the ssh host key stored in the remote source isn't correct.
You can easily check this:
* if setting the same ssh host key using the cli works fine then it's the ui that is passing a wrong host key.
* If it doesn't work also using the cli then the ssh host key is probably wrong.

Reply:
Thank you! I fixed it.

I think it's ready to merge.